jgrapht-core/src/main/java/org/jgrapht/traverse/CrossComponentIterator.java

```diff
@@ -325,7 +330,21 @@ private void addUnseenChildrenOf(V vertex)
         }
     }
 
-    private void encounterStartVertex()
+    /**
+     * Determines the outgoing edges for a given vertex.
+     *
+     * @param vertex vertex in question
+     * @return set of outgoing edges connected to the vertex
+     */
+    protected Set<E> getOutgoingEdgesOf(V vertex)
```

Comment:
I'm OK with adding this new protected method, but the name should be selectOutgoingEdges, and the Javadoc should explain its purpose.

Reply:
I'll update it and add a unit test